### PR TITLE
Concept of `autoFocus`, range/focus moved to the compat layer - recreated

### PR DESCRIPTION
**Pull request by kum-deepak**
Recreated #1790

### CODE REVIEW

File: src/base/coordinate-grid-mixin.ts [hunk 1]
```diff
@@ -1220,11 +1191,6 @@ export class CoordinateGridMixin extends ColorMixin(MarginMixin) {
         if (this._conf.zoomOutRestrict) {
             // ensure range is within self._xOriginalDomain
             range = this._checkExtents(range, this._xOriginalDomain);
-
-            // If it has an associated range chart ensure range is within domain of that rangeChart
-            if (this._rangeChart) {
```
Comment (kum-deepak) [inline src/base/coordinate-grid-mixin.ts:1225]: @gordonwoodhull this seems to be the only functional code that is getting dropped.

Comment (gordonwoodhull) [reply src/base/coordinate-grid-mixin.ts:1225]: Oh, I get it now, sorry I misunderstood.

I should not try to review PRs without trying them out!

Yes, I agree this is an obscure case. I misread your earlier comment "the earlier code was also checking the domain of the other chart and constraining based on that" and thought it meant that pan and zoom on the focus chart would not set the brush on the range chart, but of course it still does that.

Sorry again for the confusion.

Comment (kum-deepak) [reply src/base/coordinate-grid-mixin.ts:1225]: Thanks, @gordonwoodhull  for the review. :+1:

Comment (kum-deepak) [reply src/base/coordinate-grid-mixin.ts:1225]: English is my 3rd language and communication definitely not my strongest suit :smile:

**Comment by kum-deepak** [comment]
I will go ahead and merge the PR.

I have opened an issue #1793 to explore if ElasicX can play well with zoom/focus.

File: src/base/coordinate-grid-mixin.ts (hunk 1 above)
Comment (gordonwoodhull) [reply src/base/coordinate-grid-mixin.ts:1225]: dc.js is sufficiently complex that it’s not easy to write about clearly.

A lot of my own comments mystify me only months later. 🤣

**Pull request closed without merging**